src/transformStateWithClones.js

```diff
@@ -6,8 +6,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const statesArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        statesArray.push({ ...stateClone });
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateClone[key];
+        }
+        statesArray.push({ ...stateClone });
+        break;
+      case 'clear':
+        for (const key in stateClone) {
+          delete stateClone[key];
+        }
+        statesArray.push({ ...stateClone });
```

Comment:
[DONT REPEAT YOURSELF] - You're pushing a clone of `stateClone` to `statesArray` in every case of your switch statement. To adhere to the DRY principle, perform this action once after the switch block.